### DISCUSSION TITLE
Kitten context in Python REPL

### DISCUSSION THREAD

**Discussion by cmpadden** [kovidgoyal/kitty#7423] [General]
Is there a recommended approach to instantiating a Python REPL environment with access to the `kitty` and `kitten` contexts as defined when launching a kitten? (eg. `map ctrl+shift+f1 kitten script.py`). I would like to interactively develop a kitten without needing to manually invoke the script.

One solution would be to create a separate virtual environment and install kitty/kitten, but wanted to get y'alls input. Thanks!

**Comment by kovidgoyal** [comment]
There's no REPL builtin, you can however run python with either
kitty +runpy or kitty +launch no need to run it via a map. This gives
you access ot kitty and kitten packages.

Of course this wont have access to the actual live objects from a running kitty
instance, like Boss and friends.

If you want access to the objects in a live kitty instance you need to
run it via map.

**Reply by cmpadden** [reply]
This is insightful - thank you @kovidgoyal . 🙏
